app/rocket/kinetics.py

```diff
@@ -0,0 +1,90 @@
+from threading import Thread
+from collections import deque
+
+import numpy as np
+import logging
+
+WINDOW_SIZE = 50
+
+class Kinetics(Thread):
+    def __init__(self, device_factory):
+        Thread.__init__(self)
+        self.accelerometer = device_factory.accelerometer
+
+        self.time_series = deque(np.arange(WINDOW_SIZE), maxlen=WINDOW_SIZE)
+
+        self.acceleration = TimeWindow()
+        self.velocity = TimeWindow()
+        self.position = TimeWindow()
+        self.active = False
+
+    def activate(self):
+        if self.active:
+            raise Exception('Active Kinetics Model')
+        else:
+            self.active = True
+            self.start()
+
+    def deactivate(self):
+        self.active = False
+        self.join(timeout=2)
+        if self.is_alive():
+            raise Exception('Failed to deactivate Kinetics Model')
+
+    def predicted_apogee(self):
+        return 0
+
+    def acceleration(self):
+        return self.acceleration.last()
+
+    def velocity(self):
+        return self.velocity.last()
+
+    def position(self):
+        return self.position.last()
+
+    def run(self):
+        while self.active:
+            measurement = self.accelerometer.read()
+            self.time_series.append(measurement['time'])
+            self.acceleration.append(x=measurement['x'], y=measurement['y'], z=measurement['z'])
+
+            prev_velocity = self.velocity.last()
+            delta_velocity = self.acceleration.integrate_last(self.time_series[-2], self.time_series[-1])
+            self.velocity.append(
+                x=prev_velocity['x'] + delta_velocity[0],
+                y=prev_velocity['y'] + delta_velocity[1],
+                z=prev_velocity['z'] + delta_velocity[2])
+
+
+            prev_position = self.position.last()
+            delta_position = self.velocity.integrate_last(self.time_series[-2], self.time_series[-1])
+            self.position.append(
+                x=prev_position['x'] + delta_position[0],
```

Comment:
for now I will keep the x, y, z values as a hash because it makes programming it a bit easier....once this is more refined and fully tested, we can refactor it to use an array with indexes [0, 1, 2] as [x, y, z] and see what performance improvement we get